Source/WebCore/rendering/RenderThemeCocoa.mm

```diff
@@ -242,4 +243,36 @@ static inline FontSelectionValue cssWeightOfSystemFont(CTFontRef font)
 
 #endif
 
+static Color grammarColor(bool useDarkMode)
+{
+#if ENABLE(POST_EDITING_GRAMMAR_CHECKING)
+    static bool useBlueForGrammar = false;
+    static std::once_flag flag;
+    std::call_once(flag, [] {
+        useBlueForGrammar = os_feature_enabled(TextComposer, PostEditing) && os_feature_enabled(TextComposer, PostEditingUseBlueDots);
+    });
+
+    if (useBlueForGrammar)
+        return useDarkMode ? SRGBA<uint8_t> { 40, 145, 255, 217 } : SRGBA<uint8_t> { 0, 122, 255, 191 };
```

Comment:
maybe these hard-coded colors should be refactored out to `constexpr` constants for better readability?

Reply:
Some of these will start using system colors.

For the others, I don't think there's a big readability improvement, given that the resolved color is either factored into a named function, or inline with the marker type. I also can't think of names that would make the values any clearer.

Given that this is existing code, and I don't think there's a readability issue here, I will keep this as-is unless you feel strongly.